trio/_core/_asyncgens.py

```diff
@@ -170,14 +193,16 @@ async def finalize_remaining(self, runner):
         # all are gone.
         while self.alive:
             batch = self.alive
-            self.alive = set()
+            self.alive = _ASYNC_GEN_SET()
```

Comment:
I would rather use inference (`self.alive: ... = set()`) then call a thing we could change and never get told about errors about (cause it's `if not TYPE_CHECKING`).

Reply:
Just tried replacing with 
```python
self.alive: _ASYNC_GEN_SET = set()
```
and we get this error:
```console
error: Incompatible types in assignment (expression has type "set[<nothing>]", variable has type "WeakSet[AsyncGeneratorType[object, NoReturn]] | set[AsyncGeneratorType[object, NoReturn]]")  [assignment]
```

Reply:
Wait I don't understand the error. Mypy seems to think you're doing `self.alive`... oh I missed that it was `self`. Ouch.